lib/Backend/Opnd.cpp

```diff
@@ -2244,9 +2244,6 @@ IndirOpnd::New(RegOpnd *baseOpnd, RegOpnd *indexOpnd, IRType type, Func *func)
 {
     IndirOpnd * indirOpnd;
 
-    AssertMsg(baseOpnd, "An IndirOpnd needs a valid baseOpnd.");
```

Comment:
My new one covers index * scale + offset, but not base + index * scale. Technically it's only required to remove the second assert though.